### DISCUSSION TITLE
How can I retrieve the key's name from an EncryptedData node?

### DISCUSSION THREAD

**Discussion by elseifthen** [lsh123/xmlsec#464] [Q&A] [answered]
I tried to use  `xmlSecFindNode(node_ptr, xmlSecNodeKeyName, xmlSecEncNs)`  on the ` EncryptedData`  xml node (see snippet below) to get the child node ` KeyName`  whose content is the key's name stored upon encryption of the xml file. However, ` EncryptedData`  has no xml children but its content seems to be interpreted as text (probably defined by the Type attribute):
```
<EncryptedData xmlns="http://www.w3.org/2001/04/xmlenc#" Type="http://www.w3.org/2001/04/xmlenc#Content">
<EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes256-cbc"/>
<KeyInfo xmlns="http://www.w3.org/2000/09/xmldsig#">
<EncryptedKey xmlns="http://www.w3.org/2001/04/xmlenc#">
<EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#rsa-oaep-mgf1p"/>
<KeyInfo xmlns="http://www.w3.org/2000/09/xmldsig#">
<KeyName>0a40e2bc</KeyName>
</KeyInfo>
<CipherData>
<CipherValue>
…
</EncryptedData>
```
I can't figure out how to get to the content of the `KeyName` tag, i.e. how to retrieve the string `0a40e2bc` without manually parsing the text contained in the `EncryptedData` node. Can anyone help me on this?

**Comment by elseifthen** [comment, marked as answer]
Ah, I had the wrong namespace! With ` xmlSecFindNode(node_ptr, xmlSecNodeKeyName, xmlSecDSigNs)`  it works as expected.

**Reply by lsh123** [reply]
Great that you figured it out!
